zebra-scan/src/tests.rs

```diff
@@ -550,3 +509,47 @@ fn random_compact_tx(mut rng: impl RngCore) -> CompactTx {
     ctx.outputs.push(cout);
     ctx
 }
+
+/// Returns transactions belonging to any of the given [`ScanningKey`]s.
+///
+/// TODO:
+/// - Remove the `sapling_tree_size` parameter or turn it into an `Option` once we have access to
+/// Zebra's state, and we can retrieve the tree size ourselves.
+/// - Add prior block metadata once we have access to Zebra's state.
+fn scan_block<K: ScanningKey>(
```

Comment:
Optional:

I think it can be a bit confusing that our function has the same name as the ` zcash_client_backend::scanning` one. 

What about changing it to `scan_zebra_block`?

Reply:
Alternatively maybe `convert_and_scan_block()`, `scan_one_block`, or `scan_for_viewer_transactions`.

Reply:
I don't have a strong opinion here, and I'm happy to change the name, but I would prefer to keep distinguishing the two functions by their paths. More specifically, if we ever need to use the one from `zcash_client_backend`, we should explicitly prefix it with its path so it's clearer it's an external function. Also, the same names indicate the same purpose to me.

Reply:
I'm hoping it'll be moved to a module and accessed through another function later (see below), but I also prefer to distinguish them by their paths for now since they do the same thing.

```rust

mod scanning {

    use super::*;

    /// Records transactions from block belonging to any of the given [`ScanningKey`]s.
    pub fn process_block<K: ScanningKey>(
        network: Network,
        block: Arc<Block>,
        sapling_tree_size: u32,
        scanning_keys: &[&K],
    ) -> Result<(), ScanError> {
        let (network, block, keys) = args(network, block, sapling_tree_size, scanning_keys);
        let scanned_block = scan_block(network, block, keys)?;

        // TODO: record result here

        Ok(())
    }

    fn args<'a, K: ScanningKey>(
        network: Network,
        block: Arc<Block>,
        sapling_tree_size: u32,
        scanning_keys: &[&'a K],
    ) -> (
        zcash_primitives::consensus::Network,
        CompactBlock,
        Vec<(&'a AccountId, &'a K)>,
    ) {
        let network: zcash_primitives::consensus::Network = network.into();

        let chain_metadata = ChainMetadata {
            sapling_commitment_tree_size: sapling_tree_size,
            // Orchard is not supported at the moment so the tree size can be 0.
            orchard_commitment_tree_size: 0,
        };

        // Use a dummy `AccountId` as we don't use accounts yet.
        let dummy_account = AccountId::from(0);

        let scanning_keys: Vec<_> = scanning_keys
            .iter()
            .map(|&key| (&dummy_account, key))
            .collect();

        block_to_compact(block, chain_metadata);

        (
            network,
            block_to_compact(block, chain_metadata),
            scanning_keys,
        )
    }

    /// Returns transactions belonging to any of the given [`ScanningKey`]s.
    ///
    /// TODO:
    /// - Remove the `sapling_tree_size` parameter or turn it into an `Option` once we have access to
    /// Zebra's state, and we can retrieve the tree size ourselves.
    /// - Add prior block metadata once we have access to Zebra's state.
    fn scan_block<'a, K: ScanningKey>(
        network: zcash_primitives::consensus::Network,
        block: CompactBlock,
        scanning_keys: Vec<(&'a AccountId, &'a K)>,
    ) -> Result<ScannedBlock<K::Nf>, ScanError> {
        // TODO: Implement a check that returns early when the block height is below the Sapling
        // activation height.

        zcash_client_backend::scanning::scan_block(
            &network,
            block,
            &scanning_keys,
            // Ignore whether notes are change from a viewer's own spends for now.
            &[],
            // Ignore previous blocks for now.
            None,
        )
    }
}

```

Reply:
Ok, we should keep the name at least for now. I think we should move the production code here into https://github.com/ZcashFoundation/zebra/blob/main/zebra-scan/src/scan.rs